contracts/facets/FundFacet.sol

```diff
@@ -225,21 +244,39 @@ contract FundFacet is IFund {
     function payContributionOnBehalfOf(uint termId, address participant) external {
         LibFundStorage.Fund storage fund = LibFundStorage._fundStorage().funds[termId];
 
-        address currentBeneficiary = IGetters(address(this)).getCurrentBeneficiary(termId);
-
         require(
-            fund.currentState == LibFundStorage.FundStates.AcceptingContributions,
+            fund.currentState == LibFundStorage.FundStates.AcceptingContributions ||
+                fund.currentState == LibFundStorage.FundStates.CycleOngoing,
             "Wrong state"
         );
         require(fund.isParticipant[participant], "Not a participant");
-        require(currentBeneficiary != participant, "Beneficiary doesn't pay");
-        require(!fund.paidThisCycle[participant], "Already paid for cycle");
-        require(
-            !fund.isExemptedOnCycle[fund.currentCycle].exempted[participant],
-            "Participant is exempted this cycle"
-        );
 
-        _payContribution(termId, msg.sender, participant);
+        bool payNextCycle;
+
+        if (fund.currentState == LibFundStorage.FundStates.AcceptingContributions) {
```

Comment:
This code is duplicated code from payContribution. Can we optimize this like we did with joinTerm?

Reply:
Commit 5468397. Here is also the refactor to set `fund.paidThisCycle = fund.paidNextCycle` and `fund.paidNextCycle=false`